test/task.h

```diff
@@ -109,6 +109,134 @@ typedef enum {
+ } while (0)
+
+#define ASSERT_INT_BASE(a, operator, b, type, conv) \
+  ASSERT_BASE(a operator b, a, operator, b, type, conv)
+
+#define ASSERT_INT_EQ(a, b) ASSERT_INT_BASE(a, ==, b, int, i)
+#define ASSERT_INT_GE(a, b) ASSERT_INT_BASE(a, >=, b, int, i)
+#define ASSERT_INT_GT(a, b) ASSERT_INT_BASE(a, >, b, int, i)
+#define ASSERT_INT_LE(a, b) ASSERT_INT_BASE(a, <=, b, int, i)
+#define ASSERT_INT_LT(a, b) ASSERT_INT_BASE(a, <, b, int, i)
+#define ASSERT_INT_NE(a, b) ASSERT_INT_BASE(a, !=, b, int, i)
+
+#define ASSERT_UINT_EQ(a, b) ASSERT_INT_BASE(a, ==, b, unsigned int, u)
+#define ASSERT_UINT_GE(a, b) ASSERT_INT_BASE(a, >=, b, unsigned int, u)
+#define ASSERT_UINT_GT(a, b) ASSERT_INT_BASE(a, >, b, unsigned int, u)
+#define ASSERT_UINT_LE(a, b) ASSERT_INT_BASE(a, <=, b, unsigned int, u)
+#define ASSERT_UINT_LT(a, b) ASSERT_INT_BASE(a, <, b, unsigned int, u)
+#define ASSERT_UINT_NE(a, b) ASSERT_INT_BASE(a, !=, b, unsigned int, u)
+
+#define ASSERT_LONG_INT_EQ(a, b) ASSERT_INT_BASE(a, ==, b, long int, li)
+#define ASSERT_LONG_INT_GE(a, b) ASSERT_INT_BASE(a, >=, b, long int, li)
+#define ASSERT_LONG_INT_GT(a, b) ASSERT_INT_BASE(a, >, b, long int, li)
+#define ASSERT_LONG_INT_LE(a, b) ASSERT_INT_BASE(a, <=, b, long int, li)
+#define ASSERT_LONG_INT_LT(a, b) ASSERT_INT_BASE(a, <, b, long int, li)
+#define ASSERT_LONG_INT_NE(a, b) ASSERT_INT_BASE(a, !=, b, long int, li)
+
+#define ASSERT_LONG_UINT_EQ(a, b) ASSERT_INT_BASE(a, ==, b, unsigned long int, lu)
+#define ASSERT_LONG_UINT_GE(a, b) ASSERT_INT_BASE(a, >=, b, unsigned long int, lu)
+#define ASSERT_LONG_UINT_GT(a, b) ASSERT_INT_BASE(a, >, b, unsigned long int, lu)
+#define ASSERT_LONG_UINT_LE(a, b) ASSERT_INT_BASE(a, <=, b, unsigned long int, lu)
+#define ASSERT_LONG_UINT_LT(a, b) ASSERT_INT_BASE(a, <, b, unsigned long int, lu)
+#define ASSERT_LONG_UINT_NE(a, b) ASSERT_INT_BASE(a, !=, b, unsigned long int, lu)
+
+#define ASSERT_LONG_LONG_INT_EQ(a, b) ASSERT_INT_BASE(a, ==, b, long long int, lli)
+#define ASSERT_LONG_LONG_INT_GE(a, b) ASSERT_INT_BASE(a, >=, b, long long int, lli)
+#define ASSERT_LONG_LONG_INT_GT(a, b) ASSERT_INT_BASE(a, >, b, long long int, lli)
+#define ASSERT_LONG_LONG_INT_LE(a, b) ASSERT_INT_BASE(a, <=, b, long long int, lli)
+#define ASSERT_LONG_LONG_INT_LT(a, b) ASSERT_INT_BASE(a, <, b, long long int, lli)
+#define ASSERT_LONG_LONG_INT_NE(a, b) ASSERT_INT_BASE(a, !=, b, long long int, lli)
+
+#define ASSERT_LONG_LONG_UINT_EQ(a, b) ASSERT_INT_BASE(a, ==, b, unsigned long long int, llu)
+#define ASSERT_LONG_LONG_UINT_GE(a, b) ASSERT_INT_BASE(a, >=, b, unsigned long long int, llu)
+#define ASSERT_LONG_LONG_UINT_GT(a, b) ASSERT_INT_BASE(a, >, b, unsigned long long int, llu)
+#define ASSERT_LONG_LONG_UINT_LE(a, b) ASSERT_INT_BASE(a, <=, b, unsigned long long int, llu)
+#define ASSERT_LONG_LONG_UINT_LT(a, b) ASSERT_INT_BASE(a, <, b, unsigned long long int, llu)
+#define ASSERT_LONG_LONG_UINT_NE(a, b) ASSERT_INT_BASE(a, !=, b, unsigned long long int, llu)
```

Comment:
This can probably be simplified a little. It's safe to widen any signed or unsigned integral type to `int64_t` (except `uint64_t`) so only two sets of macros are needed.

For the sake of brevity, I'd call the first one `ASSERT_EQ(...)`, etc., and the second one `ASSERT_UINT64_EQ(...)`.